test/smartwallet/customSmartWallet.test.ts

```diff
@@ -0,0 +1,752 @@
+import chaiAsPromised from 'chai-as-promised';
+import { EIP712TypedData, // @ts-ignore 
+signTypedData_v4, TypedDataUtils } from 'eth-sig-util';
+import { BN, bufferToHex, toBuffer, privateToAddress } from 'ethereumjs-util';
+import { TypedRequestData, getDomainSeparatorHash, ForwardRequestType, constants, RelayRequest, ForwardRequest, RelayData } from '@rsksmart/rif-relay-common';
+import { bytes32, getTestingEnvironment, containsEvent} from '../utils/TestUtils';
+
+chai.use(chaiAsPromised);
+const assert = chai.assert;
+
+const CustomSmartWallet = artifacts.require('CustomSmartWallet');
+const SuccessCustomLogic = artifacts.require('SuccessCustomLogic');
+const TestToken = artifacts.require('TestToken');
+const TestForwarderTarget = artifacts.require('TestForwarderTarget');
+
+/**
+ * Function to get the actual token balance for an account
+ * @param token 
+ * @param account 
+ * @returns The account token balance
+ */
+async function getTokenBalance (token: TestTokenInstance, account: string): Promise<BN> {
+    return await token.balanceOf(account);
+  }
+
+  /**
+   * Function to add tokens to an account
+   * @param token 
+   * @param recipient 
+   * @param amount 
+   */
+  async function mintTokens (token: TestTokenInstance, recipient: string, amount: string): Promise<void> {
+    await token.mint(amount, recipient);
+  }
+
+  /**
+   * Function to sign a transaction
+   * @param senderPrivateKey 
+   * @param relayRequest 
+   * @param chainId 
+   * @returns  the signature and suffix data
+   */
+function signRequest(
+    senderPrivateKey: Buffer,
+    relayRequest: RelayRequest,
+    chainId: number
+): { signature: string; suffixData: string } {
+    const reqData: EIP712TypedData = new TypedRequestData(
+        chainId,
+        relayRequest.relayData.callForwarder,
+        relayRequest
+    );
+    const signature = signTypedData_v4(senderPrivateKey, { data: reqData });
+    const suffixData = bufferToHex(
+        TypedDataUtils.encodeData(
+            reqData.primaryType,
+            reqData.message,
+            reqData.types
+        ).slice((1 + ForwardRequestType.length) * 32)
+    );
+    return { signature, suffixData };
+}
+
+/**
+ * 
+ * @param request Function to create the basic relay request
+ * @param relayData 
+ * @returns The relay request with basic/default values
+ */
+function createRequest(
+    request: Partial<ForwardRequest>,
+    relayData: Partial<RelayData>
+): RelayRequest {
+    const baseRequest: RelayRequest = {
+        request: {
+            relayHub: constants.ZERO_ADDRESS,
+            from: constants.ZERO_ADDRESS,
+            to: constants.ZERO_ADDRESS,
+            value: '0',
+            gas: '1000000',
+            nonce: '0',
+            data: '0x',
+            tokenContract: constants.ZERO_ADDRESS,
+            tokenAmount: '1',
+            tokenGas: '50000'
+        },
+        relayData: {
+            gasPrice: '1',
+            domainSeparator: '0x',
```

Comment:
The domainSeparator is no longer required, according to this task https://rsklabs.atlassian.net/browse/PP-137. We have to coordinate the merge. See [this](https://github.com/rsksmart/rif-relay/pull/201/files#diff-3f69b22eb4a55fe2bec311acd54690e7a0611ea34a6f2c118717bfacbaa0579aL185)

Reply:
Yes, we rebased the branch